tests/instruments/agilent/test_agilentL4534A_with_device.py

```diff
@@ -0,0 +1,126 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import numpy as np
+
+import pytest
+from pymeasure.instruments.agilent.agilentL4534A import AgilentL4534A
+from pyvisa.errors import VisaIOError
+from pymeasure.units import ureg
+
+pytest.skip('Only work with connected hardware', allow_module_level=True)
+
+class TestAgilentL4534A:
+    """
+    Unit tests for TestAgilentL4534A class.
+
+    This test suite, needs the following setup to work properly:
+        - A TestAgilentL4534A device should be connected to the computer;
+        - The device's address must be set in the RESOURCE constant.
+    """
+
+    ###############################################################
+    # Agilent34450A device goes here:
+    RESOURCE = "TCPIP::169.254.45.32::INSTR"
```

Comment:
Please have a look here, how to make these tests skip conditionally on the definition of an adapter (which replaces your "RESOURE" constant): https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#device-tests